modules/json/src/smithy4s/http/json/SchematicJCodec.scala

```diff
@@ -929,97 +921,84 @@ private[smithy4s] class SchematicJCodec(maxArity: Int) extends Schematic[JCodecM
       override def canBeKey: Boolean = false
 
       def encodeValue(doc: Document, out: JsonWriter): Unit = doc match {
-        case DNumber(value)  => out.writeVal(value)
-        case DString(value)  => out.writeVal(value)
-        case DBoolean(value) => out.writeVal(value)
-        case DNull           => out.writeNull()
-        case DArray(values) =>
+        case s: DString  => out.writeVal(s.value)
+        case b: DBoolean => out.writeVal(b.value)
+        case n: DNumber  => out.writeVal(n.value)
+        case a: DArray =>
           out.writeArrayStart()
-          values.foreach(encodeValue(_, out))
+          a.value.foreach(encodeValue(_, out))
           out.writeArrayEnd()
-        case DObject(map) =>
+        case o: DObject =>
           out.writeObjectStart()
-          map.foreach { case (k, v) =>
-            out.writeKey(k)
-            encodeValue(v, out)
+          o.value.foreach { kv =>
+            out.writeKey(kv._1)
+            encodeValue(kv._2, out)
           }
           out.writeObjectEnd()
+        case _ => out.writeNull()
```

Comment:
is it preferable to have a `case _` or `case DNull`?

precisely, is there an impact on performance if we keep `case DNull` because it provides us with exhaustivity checking

(I'm in the process of porting those changes to my branch and I had this question

Reply:
`DNull` uses `equals` while `_` just skip any checks in runtime.

Reply:
will keep then